src/Native/System.Native/pal_mount.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_mount.h"
+#include <assert.h>
+#include <string.h>
+#include <errno.h>
+
+// Check if we should use getmntinfo or /proc/mounts
+#if HAVE_MNTINFO
+#include <sys/mount.h>
+#else
+#include <sys/statfs.h>
+#include <mntent.h>
+#define STRING_BUFFER_SIZE 8192
+#endif
+
+#if HAVE_MNTINFO
+
+int GetMountInfo(MountPointFound onFound)
+{
+    // getmntinfo returns pointers to OS-internal structs, so we don't need to worry about free'ing the object
+    int result = 0;
+    struct statfs* mounts = nullptr;
+    int count = getmntinfo(&mounts, 0);
```

Comment:
Ah, gotcha; the problem is that it is known on OS X but not Linux, so it would create a runtime different that was not abstracted by the PAL. I'll add an initial capacity that should be big enough for normal use